Workflow cicd pipeline

Hermann27 opened the pull request:
This PR adds a full CI/CD pipeline for the project, which includes the following: 

- **Node.js and npm setup**: Installs Node.js and dependencies. 
- **Test workflow**: Runs tests whenever code is pushed to 'main' or on pull requests. 
- **Deployment workflow**: Deploys the project to xyz after successful tests.

Hermann27 committed:
set up custom action yaml file that will cache dependencies

Hermann27 committed:
set up full CI/CD pipeline with test and deployment workflows
This PR adds a full CI/CD pipeline for the project, which includes the following:
- **Node.js and npm setup**: Installs Node.js and dependencies.
- **Test workflow**: Runs tests whenever code is pushed to 'main' or on pull requests.
- **Deployment workflow**: Deploys the project to xyz after tests are successful.

The workflow file is located in `.github/workflows/ci-cd-pipeline.yml`.

robstrader commented:
@Hermann27 the actions look good, but @ozzydogandev has now merged in his changes for the homepage, so now the example test config is broken. You will need to update this branch with the latest, then fix the issue. The error is that jest does not know how to handle .css files. Please see the following section of the docs: https://jestjs.io/docs/webpack#handling-static-assets

You may even want to fix that in a separate pull request, then update this PR after that one has been merged. Or you can fix it in this one.

robstrader commented:
To update your PR branch, you have 2 choices, you can rebase or merge locally, then push to the remote branch, or you can update the branch remotely. Updating remotely is fairly simple when there are no merge conflicts, you can simply click the 'Update branch' button in the PR. Then do a hard reset on your local branch to the remote branch:

![update-branch-pr](https://github.com/user-attachments/assets/718c4bcd-923c-4b3a-8390-341e238361b9)

After clicking 'Update branch':

```sh
git checkout pr-mybranch
git fetch
git reset --hard origin/pr-mybranch
```

Hermann27 committed:
Merge branch 'main' into workflow-cicd-pipeline

Hermann27 committed:
chore: update project configuration and fix linting issues

- Fixed outdated branch by merging `main` into the feature branch to sync with the latest changes.
- Resolved Jest configuration issues to properly handle CSS and style imports in tests.
- Configured ESLint to automatically fix default formatting issues such as indentation, semicolons, and spacing, ensuring consistent code style across the project.

Hermann27 commented:
PR updated 👍

robstrader commented on package.json:
```diff
@@ -6,7 +6,7 @@
     "dev": "vite",
     "build": "vite build",
-    "lint": "eslint . --report-unused-disable-directives --max-warnings 0",
+    "lint": "eslint . --fix --report-unused-disable-directives --max-warnings 0",
```
I'm not sure you actually want to use `--fix` here. You would like `lint` to report an errors. You could create a separate script for fixing named `lint:fix`.

Hermann27 replied:
I have added a separate script for fixing the lint.

robstrader commented on jest.config.mjs:
```diff
@@ -1,13 +1,21 @@
-  transform: {
+  /* transform: {
     "^.+\\.jsx?$": "babel-jest",
   },
-  moduleNameMapper: {
+ moduleNameMapper: {
     "^@/(.*)$": "<rootDir>/src/$1",
-  },
+  },*/
```
Could you go ahead and remove the commented out lines.

Hermann27 replied:
I have removed the comment as requested.

robstrader reviewed (commented):
Overall, looks great! Added a couple of comments. Good job on getting Jest working with CSS files.

Hermann27 committed:
removed comment from jest config file and added a separate script for fixing lint

robstrader reviewed (approved):
Sorry, meant to get back to this. Looks good to me, but you are going to need to merge in the latest changes from main as Ozzy has been pushing some changes.

Hermann27 committed:
no changes added to commit

Hermann27 commented:
![image](https://github.com/user-attachments/assets/b2ba2b7c-1dec-411a-93d8-ccf0a45e331a)

I have updated this PR with the latest changes as Ozzy has pushed some changes. Now I need to fix all those eslint error

Hermann27 commented:
for now, I have turned off these rules:

- "react/prop-types": "off",

- "react/react-in-jsx-scope": "off",

- "react/no-unescaped-entities": "off",

What would be the best way to correct those errors?
